### PR TITLE
stat init

### PR DESCRIPTION
**Pull request by kzoper**
another try

### CODE REVIEW

File: stat/stat.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package stat
+
+import (
+	"github.com/rocketlaunchr/dataframe-go"
+	"math"
+)
+
+type Stat interface {
```
Comment (pjebs) [inline stat/stat.go]: SeriesRational interface {

Comment (rocketlaunchr-cto) [inline stat/stat.go]: type SeriesRational interface {

File: stat/stat.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	dataframe.Series
+	Abs()
+}
+
+type StFloat64 struct {
```
Comment (rocketlaunchr-cto) [inline stat/stat.go]: type SeriesFloat64 struct {

File: stat/stat.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	*dataframe.SeriesFloat64
+}
+
+type SeriesInt64 struct {
+	*dataframe.SeriesInt64
+}
+
+func (s *StFloat64) Abs() {
+	for key, value := range s.Values {
+		val := math.Abs(*value)
```
Comment (rocketlaunchr-cto) [inline stat/stat.go]: check value is nil

File: stat/stat.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		s.Values[key] = &val
+	}
+}
+
+func (s *SeriesInt64) Abs() {
+	for key, value := range s.Values {
+		val := *value
```
Comment (rocketlaunchr-cto) [inline stat/stat.go]: checkif value is nil

File: stat/stat_test.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"fmt"
+	"github.com/rocketlaunchr/dataframe-go"
+	"testing"
+)
+
+func TestSeriesFloat64_Abs(t *testing.T) {
+	s1 := StFloat64{dataframe.NewSeriesFloat64("num",nil,-1.1,1.1)}
+	s1.Abs()
+	fmt.Println(s1.String())
```
Comment (rocketlaunchr-cto) [inline stat/stat_test.go]: fmt.Println will automatically call the object's `String()` method.

File: stat/stat_test.go [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package stat
+
+import (
+	"fmt"
+	"github.com/rocketlaunchr/dataframe-go"
+	"testing"
+)
+
+func TestSeriesFloat64_Abs(t *testing.T) {
```
Comment (rocketlaunchr-cto) [inline stat/stat_test.go]: The test should properly test if the results are what are expected

File: stat/stat_test.go [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package stat
+
+import (
+	"fmt"
```
Comment (rocketlaunchr-cto) [inline stat/stat_test.go]: standard packages go at top then new line, then internal packages then new line then external a pacakges.

**Comment by kzoper** [comment]
Please stop write messages like in chat or create chat in discord or gitter. 

This function is just for test, how it will work. 
Isn't work
```golang
package main

import (
	"fmt"
	"github.com/kzoper/dataframe-go"
	"github.com/kzoper/dataframe-go/stat"
)

func main() {
	s2 := dataframe.NewSeriesFloat64("sales", nil, 50.3, -23.4, 56.2, nil, nil, 84.2, 72, 89)
	stat.StFloat64{SeriesFloat64: s2}.Abs()
	fmt.Println(s2)
}

```

**Comment by rocketlaunchr-cto** [comment]
Github automatically makes it like chat format. I just wrote "single comments" in the "Files Changed" tab. They end up in the "conversations" section as "chat like" comments.

**Comment by rocketlaunchr-cto** [comment]
The main issue is in your `stat` package, you are referencing `"github.com/rocketlaunchr/dataframe-go"`, but in your main file, you are referencing `"github.com/kzoper/dataframe-go"` so they are different types.

I also see what you are trying to do. In that case remove the pointers and pointer references:

```
type StFloat64 struct {
	dataframe.SeriesFloat64
}

type SeriesInt64 struct {
	dataframe.SeriesInt64
}

func (s StFloat64) Abs() {
	for key, value := range s.Values {
		if value == nil {
			continue
		}
		val := math.Abs(*value)
		s.Values[key] = &val
	}
}

func (s SeriesInt64) Abs() {
	for key, value := range s.Values {
		val := *value
		if val < 0 {
			val = val * -1
		}
		s.Values[key] = &val
	}
}
```
In your `main.go`:

```
	s2 := dataframe.NewSeriesFloat64("sales", nil, 50.3, -23.4, 56.2, nil, nil, 84.2, 72, 89)
	stat.StFloat64{*s2}.Abs()
	fmt.Println(s2)
```

**Comment by kzoper** [comment]
No. Change reference and : 

``` golang
package main

import (
	"fmt"
	"github.com/kzoper/dataframe-go"
	"github.com/kzoper/dataframe-go/stat"
)

func main() {
	//s1 := dataframe.NewSeriesInt64("day", nil, 1, 2, 3, 4, 5, 6, 7, 8)
	s2 := dataframe.NewSeriesFloat64("sales", nil, 50.3, -23.4, 56.2, nil, nil, 84.2, 72, 89)
	//df := dataframe.NewDataFrame(s1, s2)
	stat.StFloat64{SeriesFloat64: s2}.Abs() // error

	fmt.Println(s2)
}

```

./main.go:13:35: cannot call pointer method on stat.StFloat64 literal
./main.go:13:35: cannot take the address of stat.StFloat64 literal

**Comment by rocketlaunchr-cto** [comment]
See my prior comment:

1) In stat package - remove pointers and pointer receivers in the your functions (sorry for misleading you. I now have a better understanding of what you are trying to do)
2) `stat.StFloat64{*s2}.Abs()` // dereference the pointer

**Comment by kzoper** [comment]
```golang
package stat

import (
	"github.com/kzoper/dataframe-go"
	"math"
)

type Stat interface {
	dataframe.Series
	Abs()
}

type StFloat64 struct {
	dataframe.SeriesFloat64
}

type SeriesInt64 struct {
	dataframe.SeriesInt64
}

func (s *StFloat64) Abs() {
	for key, value := range s.Values {
		val := math.Abs(*value)
		s.Values[key] = &val
	}
}

func (s *SeriesInt64) Abs() {
	for key, value := range s.Values {
		val := *value
		if val < 0 {
			val = val * -1
		}
		s.Values[key] = &val
	}
}
```
```golang
// it remove pointer here isn't work too
func (s StFloat64) Abs() {
	for key, value := range s.Values {
		val := math.Abs(*value)
		s.Values[key] = &val
	}
}
```
change. Still isn't working

**Comment by rocketlaunchr-cto** [comment]
You have to do a nil check because some of your values are nil. See the image I attached.

**Comment by kzoper** [comment]
Now it works

**Comment by rocketlaunchr-cto** [comment]
Why did you close the PR?

**Comment by kzoper** [comment]
1. Because Abs is only one function that change values. Other function return series of bools or one value.
2.  Using stat.StFloat64{*s2} inconveniently. We have to find another way.
screenshot from 2018-11-04 04-28-49

**Pull request closed without merging**